server/src/main/scala/services/OfferHandler.scala

```diff
@@ -0,0 +1,43 @@
+package services
+
+import akka.stream.QueueOfferResult.Enqueued
+import akka.stream.scaladsl.SourceQueueWithComplete
+import org.slf4j.{Logger, LoggerFactory}
+
+import scala.concurrent.Await
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+import scala.language.postfixOps
+import scala.util.{Failure, Success, Try}
+
+object OfferHandler {
+  val log: Logger = LoggerFactory.getLogger(getClass)
+
+  def offerWithRetries[A](queue: SourceQueueWithComplete[A], thingToOffer: A, retries: Int): Unit = {
```

Comment:
Should this return a Future of Unit? Is this blocking otherwise? Does it need to be?

Reply:
I've replaced this retry code with one from a github gist which is a lot smarter and smaller!